### DISCUSSION TITLE
What does VOLTAGE_DROP exactly?

### DISCUSSION THREAD

**Discussion by wetty** [Louisvdw/dbus-serialbattery#822] [Q&A]
Hello,
my System is a Multiplus II 5000 with two 18s packs with JKBMS.
SVS - Shared Voltage Sense is off.

Multiplus shows 62.1V which is exactly what  I want (18 x 3.45V)
The JKBMS both show 61.92V which is, after measuring it, the correct Value! So Batteries should be charged a little bit more to reach 62.1V

So I to set VOLTAGE_DROP to 0.18V

After restarting the system the values are exact the same.  
My question is now, does serialbattery think that the JKBMS Value 61.92V plus the voltage drop 0.18V ist the correct value? And with this switch to float?

Currently the only way to get JKBMS/my Batteries to 62.1V, is to switch SVS on.

Or should I change the settings in my Multiplus to set it to a higher voltage?

**Comment by mr-manuel** [comment]
Check this https://github.com/Louisvdw/dbus-serialbattery/blob/master/etc/dbus-serialbattery/battery.py and search dor `VOLTAGE_DROP`. If you do not understand it from the code I can explaint it to you.

**Reply by wetty** [reply]
(utils.MAX_CELL_VOLTAGE * self.cell_count) - utils.VOLTAGE_DROP

Which is the behaviour I thought it is
"does serialbattery think that the JKBMS Value 61.92V plus the voltage drop 0.18V ist the correct value"

Multiplus shows 62.1V, JKBMS shows 61.92V
18x3.45 - 0.18V = 61,92V which is the voltage serialbattery switches to float, when we trust the multiplus voltage measurement.

What can I do when I want the JKBMS Voltage to be the correct one?

**Reply by mr-manuel** [reply]
You have to divide your voltage drop by the cell count and add that to the max cell voltage. I will check, if we can change this somehow, since it should be added in order that the BMS gets the final voltage.

**Reply by wetty** [reply]
So voltage drop will be added in front, to get a higher V from The Multiplus, and den subtract, that 62.1 V is the correct Voltage on the BMS.
Will see to try this tomorrow.

**Reply by mr-manuel** [reply]
Currently not, but that would be the correct way. You could test to enter a negative value with the current version.

**Reply by wetty** [reply]
Negative Value will not work, as the victron sticks to 62.1 which is only 61.92V on the bms. Negative only increase the value the bms should get. 
I increase my call voltage to 3.46 (+ 0.01) and add a voltage drop of 0.18 (18 cells * 0.01) 
The next 3-4 hours will show whether this works

**Reply by mr-manuel** [reply]
I made some changes to how the `VOLTAGE_DROP` behaves. Now it should be correct. See https://github.com/mr-manuel/venus-os_dbus-serialbattery/commit/36ff1b05d7257a2902c81deb290d64fe90c50441

Could you please test https://github.com/mr-manuel/venus-os_dbus-serialbattery/releases/tag/v1.0.20230927beta and read the description in the `config.default.ini` to check, if it's understandable and if not make recommendations?

**Reply by wetty** [reply]
Its installed. Charge Voltage now shows the value plus the voltage drop. Which is correct.
Description is understandable for me, but with the problem that I know what the problem is :-)
But should be understandable in general

**Reply by mr-manuel** [reply]
I just rebuild the release since it contained a little typo, in case you want to download and install it. The version number is the same.

**Reply by wetty** [reply]
Your Version works like expected :-) Perfect!
I just have to increase to voltage drop a little bit to finally reach 62.1V. But maybe I will first wait another hour to see  if it is reached.
See my values in a screenshot from bei homeassistant Dashboard
![Bildschirmfoto 2023-09-27 um 12 54 26](https://github.com/Louisvdw/dbus-serialbattery/assets/1732929/5628fffb-d151-444e-8757-6d49ace7717e)

**Reply by wetty** [reply]
Slightly increase the voltage drop and it looks perfect. Just waiting for the switch to float Voltage 
![Bildschirmfoto 2023-09-27 um 13 38 22](https://github.com/Louisvdw/dbus-serialbattery/assets/1732929/08b80410-54db-4690-a026-1bedbd029555)

**Reply by wetty** [reply]
First Battery switched to float mode. Mission Completed!

Thanks a lot.

**Reply by mr-manuel** [reply]
You're welcome
